fluent-package/templates/package-scripts/fluent-package/deb/prerm

```diff
@@ -16,18 +16,52 @@ set -e
 # for details, see https://www.debian.org/doc/debian-policy/ or
 # the debian-policy package.
 
+make_sure_working_dir_exists() {
+    # This func is for CI ONLY.
+    # Some CI cases don't provide env with systemd.
+    # tmpfiles.d owns /tmp/<%= package_dir %>, but not created without systemd
+    if [ -d /run/systemd/system ]; then
+        return
+    fi
+    mkdir -p "/tmp/<%= package_dir %>"
+}
 
-case "$1" in
-    remove|upgrade|deconfigure)
-    ;;
+leave_info_for_auto_restart_if_need() {
+    # Some CI cases that don't provide env with systemd, so disable auto restart feature in those cases.
+    if [ ! -d /run/systemd/system ]; then
+        return
+    fi
 
-    failed-upgrade)
-    ;;
+    pid="$(systemctl show "<%= service_name %>" --property=MainPID --value)"
+    if [ $pid -eq 0 ]; then
+        echo "Do not use auto restart because the service is not active"
+        return
+    fi
+
+    . "/etc/default/<%= service_name %>"
+    echo "FLUENT_PACKAGE_SERVICE_RESTART: $FLUENT_PACKAGE_SERVICE_RESTART"
+    if [ "$FLUENT_PACKAGE_SERVICE_RESTART" != auto ]; then
+        return
+    fi
```

Comment:
Could you comment a possibility that there is a case that these temporary files are left without clean-up.

Reply:
Opps, above comment is nonsense. sorry.